lib/hacluster.pm

```diff
@@ -524,4 +526,20 @@ sub activate_ntp {
     systemctl "enable --now $ntp_service.service";
 }
 
+sub gen_root_ssh_key {
+    # Generate an ssh key for root
+    # Was done before by 'ha-cluster-init', see bsc#1169581 for more informations
+    assert_script_run 'ssh-keygen -t rsa -f /root/.ssh/id_rsa -N "" <<< y';
+    assert_script_run 'cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys';
+}
+
+sub get_root_ssh_key {
+    my $node_to_join = shift;
+
+    # Configure ssh key to enable ssh passwordless
+    add_to_known_hosts($node_to_join);
+    exec_and_insert_password("scp root\@$node_to_join:/root/.ssh/* /root/.ssh/");
+    assert_script_run "ssh root\@$node_to_join 'ssh-keyscan -H ' . get_hostname . ' >> /root/.ssh/known_hosts'";
```

Comment:
The get_hostname doesn't work in the ssh call. I believe I had the same problem while I was doing the yast cluster test.
However, this solution is working in my [lab](http://1a102.qa.suse.de/tests/4361#step/ha_cluster_join/8) using the build with the bad crmsh patch:
```
sub get_root_ssh_key {
    my $node_to_join = shift;
    my $hostname     = get_hostname;
    # Configure ssh key to enable ssh passwordless
    add_to_known_hosts($node_to_join);
    exec_and_insert_password("scp root\@$node_to_join:/root/.ssh/* /root/.ssh/");
    assert_script_run "ssh -o StrictHostKeyChecking=no root\@$node_to_join 'ssh-keyscan -H $hostname >> /root/.ssh/known_hosts'";
}
```

Reply:
Indeed. With `get_hostname` between double quotes, the method is not going to be called. Instead, the string `get_hostname` itself will be part of the shell command. Good catch!

Another option is to call it like this:
```
assert_script_run "ssh root\@$node_to_join 'ssh-keyscan -H '" . get_hostname . "' >> /root/.ssh/known_hosts'";
```
But I think it looks better with the `$hostname` variable.

Reply:
Thanks! I'm doing the change.